crates/wit-bindgen-rust/src/interface.rs

```diff
@@ -2271,27 +2270,37 @@ impl<'a> wit_bindgen_core::InterfaceGenerator<'a> for InterfaceGenerator<'a> {
         self.rustdoc(docs);
         let camel = to_upper_camel_case(name);
 
+        let wrpc_transport = self.gen.wrpc_transport_path();
         let anyhow = self.gen.anyhow_path();
         let async_trait = self.gen.async_trait_path();
         let bytes = self.gen.bytes_path();
         let futures = self.gen.futures_path();
         let tracing = self.gen.tracing_path();
-        let wrpc_transport = self.gen.wrpc_transport_path();
 
         if self.in_import {
             uwriteln!(
                 self.src,
                 r#"
+#[derive(Debug, Clone, Copy)]
+pub struct {camel}Rep;
+
+#[derive(Debug)]
+pub struct {camel} {{
+    pub handle: {wrpc_transport}::ResourceOwn<{camel}Rep>
+}}
+
 #[derive(Debug)]
-pub struct {camel}(String);
+pub struct {camel}Borrow {{
+    pub handle: {wrpc_transport}::ResourceBorrow<{camel}Rep>
+}}
```

Comment:
That would definitely be ideal! The reason it's like this for now is that `wit-bindgen-rust` can impl import methods directly on `{wrpc_transport}::Resource{Own,Borrow}<T>` since it's a foreign type. Do you think having it define a separate `{camel}Client` struct that it impl's the imported resource methods would make sense?

Reply:
I see what you mean, we can probably use an "extension trait" trick here. Something like:
```
pub trait MyResource {
   fn mymethod() -> impl Future<Output=anyhow::Result<()>>
}

impl MyResource for wrpc_transport::ResourceOwn<&dyn MyResource> {}
impl MyResource for wrpc_transport::ResourceBorrow<&dyn MyResource> {}
```

Let's just keep it as it is for now and explore how we can improve this later.